### PR TITLE
Add Blob Download functionality to the connector

### LINKED ISSUE #5516: Provide download blob support in msazurestorage
**Description:**
WSO2 don't have download blob support in msazurestorage. Better if this can be addressed in a future release of msazurestorage connector.
Further more when adding the GET Request to download a file better if can reconsider to add the HEAD Reqeust to get the metadata of a file

**Suggested Labels:**

**Suggested Assignees:**

**Affected Product Version:**

**OS, DB, other environment details and versions:**    

**Steps to reproduce:**

**Related Issues:**

### PR DESCRIPTION
**Pull request by juanchiriera**
## Purpose
A specific sequence for handling file downloads was added. It will allow the user to specify the container and the file to be downloaded. 
Fixes https://github.com/wso2/product-ei/issues/5516

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -200,6 +200,11 @@
             <artifactId>azure-storage</artifactId>
             <version>${azure.version}</version>
         </dependency>
+        <dependency>
+            <groupId>org.apache.synapse</groupId>
+            <artifactId>synapse-nhttp-transport</artifactId>
+            <version>2.1.7-wso2v19</version>
```
Comment (GDLMadushanka) [inline pom.xml]: please move the dependency version to the properties section of the pom

File: src/main/java/org/wso2/carbon/connector/azure/storage/BlobDownloader.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        OMFactory factory = OMAbstractFactory.getOMFactory();
+        OMNamespace ns = factory.createOMNamespace(AzureConstants.AZURE_NAMESPACE, AzureConstants.NAMESPACE);
+        OMElement result = factory.createOMElement(AzureConstants.RESULT, ns);
+        ResultPayloadCreator.preparePayload(messageContext, result);
+        try {
+	        String storageConnectionString = AzureUtil.getStorageConnectionString(messageContext);
+	        CloudStorageAccount account = CloudStorageAccount.parse(storageConnectionString);
+	        CloudBlobClient serviceClient = account.createCloudBlobClient();
+
+            CloudBlobContainer container = serviceClient.getContainerReference((String) containerName);
```
Comment (GDLMadushanka) [inline src/main/java/org/wso2/carbon/connector/azure/storage/BlobDownloader.java:70]: fix formatting issue

File: src/main/java/org/wso2/carbon/connector/azure/storage/BlobDownloader.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+	        
+	        String contentType = handler.getContentType();
+	        org.apache.axis2.context.MessageContext axis2MsgCtx = 
+	        		((org.apache.synapse.core.axis2.Axis2MessageContext) messageContext)
+	        		.getAxis2MessageContext();
+	        Builder builder = BuilderUtil.getBuilderFromSelector(contentType, axis2MsgCtx);
+           
+                      	
+           	JsonUtil.removeJsonPayload(axis2MsgCtx);
+            axis2MsgCtx.
```
Comment (GDLMadushanka) [inline src/main/java/org/wso2/carbon/connector/azure/storage/BlobDownloader.java]: fix formatting

File: src/main/java/org/wso2/carbon/connector/azure/storage/BlobDownloader.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+        } catch (InvalidKeyException e) {
+            handleException("Invalid account key found.", e, messageContext);
+        } catch (StorageException e) {
+            handleException("Error occurred while connecting to the storage.", e, messageContext);
+        } catch (NoSuchElementException e) {
+            // No such element exception can be occurred due to server authentication failure.
+            handleException("Error occurred while listing the container", e, messageContext);
+        } catch (ConnectException e) {
+            handleException("Unexpected error occurred.", e, messageContext);
+		} catch (IOException e) {
```
Comment (GDLMadushanka) [inline src/main/java/org/wso2/carbon/connector/azure/storage/BlobDownloader.java]: fix formatting

File: pom.xml (hunk 1 above)
Comment (juanchiriera) [reply pom.xml]: Resolved on latest commit.

File: src/main/java/org/wso2/carbon/connector/azure/storage/BlobDownloader.java (hunk 2 above)
Comment (juanchiriera) [reply src/main/java/org/wso2/carbon/connector/azure/storage/BlobDownloader.java:70]: Resolved on latest commit

File: src/main/java/org/wso2/carbon/connector/azure/storage/BlobDownloader.java (hunk 3 above)
Comment (juanchiriera) [reply src/main/java/org/wso2/carbon/connector/azure/storage/BlobDownloader.java]: Resolved on latest commit.

File: src/main/java/org/wso2/carbon/connector/azure/storage/BlobDownloader.java (hunk 4 above)
Comment (juanchiriera) [reply src/main/java/org/wso2/carbon/connector/azure/storage/BlobDownloader.java]: Resolved on latest commit.

**Review by GDLMadushanka — APPROVED** [review]

**Pull request merged**